src/charwise.rs

```diff
@@ -908,8 +911,10 @@ impl CharwiseDoubleArrayAhoCorasick {
     #[inline(always)]
     unsafe fn get_next_state_id_leftmost_unchecked(&self, mut state_id: u32, c: char) -> u32 {
         loop {
-            if let Some(state_id) = self.get_child_index_unchecked(state_id, c) {
-                return state_id;
+            if let Some(mapped_c) = self.mapper.get(c) {
+                if let Some(state_id) = self.get_child_index_unchecked(state_id, mapped_c) {
+                    return state_id;
+                }
```

Comment:
```rust
let mapped_c_opt = self.mapper.get(c);
loop {
    if let Some(mapped_c) = mapped_c_opt {
```
Computation of`self.mapper.get(c)` can be placed outside the loop.

Reply:
I fixed get_next_state_id_unchecked(), but I don't want to do that for get_next_state_id_leftmost_unchecked() because that variable worsens readability.
Modern compilers will do that automatically because `self` and `c` are immutable.

Reply:
Okay